bazel/BUILD

```diff
@@ -286,6 +286,11 @@ config_setting(
     values = {"define": "deprecated_features=disabled"},
 )
 
+config_setting(
```

Comment:
Let's use `bool_flag` going forward instead of `define`. Some example usage are in Lua and all extension enable/disable:
https://github.com/envoyproxy/envoy/blob/v1.17.1/source/extensions/filters/common/lua/BUILD#L13-L23

Reply:
Done!